bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/WrappedScheduledExecutorService.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2014,2017 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.common;
+
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class wraps the ScheduledThreadPoolExecutor to implement the {@link #afterExecute(Runnable, Throwable)} method
+ * and log the error. The error will otherwise go unnoticed because on an exception thrown in the thread it will simply
+ * end with no logging or whatsoever unless the user handles it. This wrappers removes the burden by the user to always
+ * catch errors in scheduled threads for logging and it also catches unhandled exceptions that can be the cause of very
+ * hard to catch bugs because no error is ever shown.
+ * The implementation follows the implementation suggestion as given in the JavaDoc of the method.
```

Comment:
Hm, I assume this line can / should be removed.
We don't need to state that our implementation follows an implementation suggestion or that we do it correctly with respect to something.
WDYT?